src/main/java/com/weirddev/testme/intellij/intention/CreateTestForMethodIntention.java

```diff
@@ -0,0 +1,90 @@
+import com.intellij.openapi.editor.Editor;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.ui.Messages;
+import com.intellij.psi.*;
+import com.intellij.util.IncorrectOperationException;
+import com.weirddev.testme.intellij.action.CreateTestMeAction;
+import com.weirddev.testme.intellij.action.TestMeActionHandler;
+import com.weirddev.testme.intellij.builder.MethodFactory;
+import org.jetbrains.annotations.NonNls;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * create test for selected method
+ *
+ * @author huangliang
+ */
+@NonNls
+final class CreateTestForMethodIntention extends PsiElementBaseIntentionAction implements IntentionAction {
+
+    /**
+     * Checks whether this intention is available at the caret offset in file - the caret must sit a testable method
+     * character in a ternary statement. If this condition is met, this intention's entry is shown in the available
+     * intentions list.
+     *
+     * <p>Note: this method must do its checks quickly and return.</p>
+     *
+     * @param project a reference to the Project object being edited.
+     * @param editor  a reference to the object editing the project source
+     * @param element a reference to the PSI element currently under the caret
+     * @return {@code true} if the caret is in a literal string element, so this functionality should be added to the
+     * intention menu or {@code false} for all other types of caret positions
+     */
+    public boolean isAvailable(@NotNull Project project, Editor editor, @Nullable PsiElement element) {
+        // Quick sanity check
+        return null != element && element instanceof PsiIdentifier && element.getParent() instanceof PsiMethod;
+    }
+
+    /**
+     * pop up the testMe menu list
+     * @param project a reference to the Project object being edited.
+     * @param editor  a reference to the object editing the project source
+     * @param element a reference to the PSI element currently under the caret
+     * @throws IncorrectOperationException Thrown by underlying (PSI model) write action context
+     *                                     when manipulation of the PSI tree fails.
+     */
+    public void invoke(@NotNull Project project, Editor editor, @NotNull PsiElement element)
+        throws IncorrectOperationException {
+        PsiMethod method = (PsiMethod)element.getParent();
+        final PsiClass srcClass = CreateTestMeAction.getContainingClass(element);
+        // if is not testable show a error dialog
+        if (!MethodFactory.isTestable(method, srcClass)) {
+            String message = "Method \"" + method.getName() + "\" is not testable!";
+            String title = "TestMe: Method Not Testable";
+            Messages.showErrorDialog(message, title);
+        } else {
+            new TestMeActionHandler(true).invoke(project, editor, element.getContainingFile());
```

Comment:
would probably be better to pass the method element already resolved by IDEA rather then pass the file and try to resolve it by cursor location.  we can create another method for this use case on TestMeActionHandler

Reply:
Furthermore, there's still a bug when trying to generate another test class (not via intention for method) for a class that already has a test, if cursor is on a test method - existing test is updated instead of popping a window to select alternative test class name as in previous behavior. there are also other exceptions and errors. my point is, that now that method generation is explicitly triggered by user via intention - we need to take advantage of this and separate logic in both CreateTestMeAction and TestMeActionHandler for these 2 user scenarios . this includes changes already been merged in previous PR.